shotover-proxy/src/transforms/cassandra/sink_cluster.rs

```diff
@@ -336,6 +379,35 @@ fn get_nodes_from_system_peers(
     new_nodes
 }
 
+fn is_use_statement(request: &mut Message) -> bool {
+    if let Some(Frame::Cassandra(frame)) = request.frame() {
+        // CassandraOperation::Error(_) is another possible case, we should silently ignore such cases
+        if let CassandraOperation::Query { query, .. } = &mut frame.operation {
+            if let CassandraStatement::Use(_) = query.as_mut() {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+fn is_use_statement_successful(response: Option<Result<Response>>) -> bool {
+    if let Some(Ok(Response {
+        response: Ok(mut response),
+        ..
+    })) = response
+    {
+        if let Some(Frame::Cassandra(CassandraFrame {
+            operation: CassandraOperation::Result(CassandraResult::SetKeyspace(_)),
```

Comment:
Should we be checking that the keyspace set was the one we expect?

Reply:
Are there any known cases where that could occur?
I feel its safe to assume that cassandra does its documented to do.

If the assumption could lead to a shotover panic then I would be more inclined to double check cassandra is behaving properly.

Reply:
I was imagining some sort of race condition that with multiple consecutive `USE` statements but I think you're right and we should be right with this.